packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -1312,7 +1318,7 @@ class _CupertinoActionSheetActionState extends State<CupertinoActionSheetAction>
         + fontSize * _kActionSheetButtonVerticalPaddingFactor;
 
     return MouseRegion(
-      cursor: kIsWeb ? SystemMouseCursors.click : MouseCursor.defer,
+      cursor: kIsWeb ? SystemMouseCursors.click : widget.cursor,
```

Comment:
Can the cursor be configured to accept `WidgetStateMouseCursor`s? It would be nice to have that functionality added right off the bat.

Reply:
would like some help with this... not sure how to exactly approach it

would it be similar to this pr: https://github.com/flutter/flutter/pull/151788/files

Reply:
Is this better:
```dart
final WidgetStateProperty<MouseCursor> effectiveMouseCursor =
    WidgetStateProperty.resolveWith<MouseCursor>((Set<WidgetState> states) {
      return WidgetStateProperty.resolveAs<MouseCursor?>(widget.mouseCursor, states)
          ?? (kIsWeb && !states.contains(WidgetState.disabled)
              ? SystemMouseCursors.click
              : SystemMouseCursors.basic
          );
    });

    return MouseRegion(
      cursor: effectiveMouseCursor.resolve(<WidgetState>{}),
      child: MetaData(
        metaData: this,
        behavior: HitTestBehavior.opaque,
        child: ConstrainedBox(
          constraints: const BoxConstraints(
            minHeight: _kActionSheetButtonMinHeight,
          ),
          child: Semantics(
            button: true,
            onTap: widget.onPressed,
            child: Padding(
              padding: EdgeInsets.fromLTRB(
                _kActionSheetButtonHorizontalPadding,
                verticalPadding,
                _kActionSheetButtonHorizontalPadding,
                verticalPadding,
              ),
              child: DefaultTextStyle(
                style: style,
                textAlign: TextAlign.center,
                child: Center(child: widget.child),
              ),
            ),
          ),
        ),
      ),
    );

Reply:
Only the `resolveAs` block is needed. The wrapping with `resolveWith` was done in the other PR because the  `buildToggleable` method required its mouse cursor to be a `WidgetStateProperty<MouseCursor>` instead of just a `MouseCursor`. But that is not the case here with `MouseRegion`.

Reply:
```dart
final MouseCursor effectiveMouseCursor =
        WidgetStateProperty.resolveAs<MouseCursor?>(widget.mouseCursor, states)
            ?? (kIsWeb && !states.contains(WidgetState.disabled)
            ? SystemMouseCursors.click
            : SystemMouseCursors.basic
        );

    return MouseRegion(
      cursor: effectiveMouseCursor,
```
just a snippet of the code since everything else is pretty much the same

Reply:
I'm not sure CupertinoActionSheetAction has a `states` getter. So checking that an action is not disabled instead should be sufficient.

I recommend playing around with it locally and writing tests until they pass. It's easier to review commits vs comment snippets here and I personally find that playing with a system on my own vastly improves my understanding of it.

Reply:
makes sense, let me play around more and see what best way to do it and will update the pr accordingly.

Thanks

Reply:
marking as resolved taken care in recent pr